pennylane/ops/qubit/special_unitary.py

```diff
@@ -0,0 +1,346 @@
+
+        if len(theta_shape) not in {1, 2}:
+            raise ValueError(
+                "Expected the parameters to have one or two dimensions without or with "
+                f"broadcasting, respectively. The parameters have shape {theta_shape}"
+            )
+
+        if theta_shape[-1] != expected_dim:
+            raise ValueError(
+                f"Expected the parameters to have shape ({expected_dim},) or (batch_size, "
+                f"{expected_dim}). The parameters have shape {theta_shape}"
+            )
+
+        super().__init__(theta, wires=wires, do_queue=do_queue, id=id)
+
+    @staticmethod
+    def compute_matrix(theta, num_wires):
+        r"""Representation of the operator as a canonical matrix in the computational basis
+        (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.SpecialUnitary.matrix`
+
+        Args:
+            theta (tensor_like): Pauli coordinates of the exponent :math:`A(\theta)`
+            num_wires (int): The number of wires
+
+        Returns:
+            tensor_like: canonical matrix
+
+        .. note::
+
+            Note that this method internally handles a complex-valued tensor of size
+            ``(4**num_wires, 2**num_wires, 2**num_wires)``, which requires at least
+            ``4 ** (2 * num_wires - 13)`` GB of memory (at default precision).
+
+        **Example**
+
+        >>> theta = np.array([0.5, 0.1, -0.3])
+        >>> qml.SpecialUnitary.compute_matrix(theta, num_wires=1)
+        array([[ 0.83004499-0.28280371j,  0.0942679 +0.47133952j],
+               [-0.0942679 +0.47133952j,  0.83004499+0.28280371j]])
+        """
+        return special_unitary_matrix(theta, num_wires)
+
+    @staticmethod
+    def compute_decomposition(theta, wires, num_wires):
+        r"""Representation of the operator as a product of other operators (static method).
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+        This ``Operation`` is decomposed into the corresponding ``QubitUnitary``.
+
+        .. seealso:: :meth:`~.SpecialUnitary.decomposition`.
+
+        Args:
+            theta (tensor_like): Pauli coordinates of the exponent :math:`A(\theta)`
+            wires (Iterable[Any] or Wires): the wire(s) the operation acts on
+            num_wires (int): The number of wires
+
+        Returns:
+            list[Operator]: decomposition of the operator
+
+        **Example:**
+
+        >>> theta = np.array([0.5, 0.1, -0.3])
+        >>> qml.SpecialUnitary.compute_decomposition(theta, 0, num_wires=1)
+        [QubitUnitary(array([[ 0.83004499-0.28280371j,  0.0942679 +0.47133952j],
+            [-0.0942679 +0.47133952j,  0.83004499+0.28280371j]]), wires=[0])]
+        """
+        return [qml.QubitUnitary(special_unitary_matrix(theta, num_wires), wires=wires)]
```

Comment:
So QubitUnitary knows how to decompose an arbitrary matrix? Wasnt aware of that! 😮

Reply:
Only for one and two qubits :grimacing: